### PR TITLE
bumped to 0.7.0, updated README and change-notes (#525)

### LINKED ISSUE #525: release 0.7.0

### PR DESCRIPTION
**Pull request by adietish**
fixes #525

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ It is available for install from the [JetBrains Marketplace](https://plugins.jet
  - Navigate into Kubernetes workloads, services, pods and nodes
  - Switching contexts and namespaces within Kubernetes
  - Follow logs from the connected cluster resource
+ - Open terminal to containers running on the connected cluster
```
Comment (mohitsuman) [inline README.md:26]:
```suggestion
 - Open a Terminal Shell to a Running Container in the connected cluster
```

File: README.md [hunk 2]
```diff
@@ -112,13 +113,21 @@ The very same dialog may also be opened with a link that's present in the notifi
 #### Delete Resources
 You may delete any resource that is listed in the resource tree by choosing "Delete" in the context menu.
 
-#### Follow logs
+#### Follow Logs
 
 IntelliJ-Kubernetes extension allows you to follow logs of the cluster resources such as Pods, Jobs and Builds.
```
Comment (mohitsuman) [inline README.md:118]:
```suggestion
IntelliJ Kubernetes extension allows you to follow logs of the cluster resources such as Pods, Jobs and Builds.
```

File: README.md [hunk 3]
```diff
@@ -112,13 +113,21 @@ The very same dialog may also be opened with a link that's present in the notifi
...
 There is a dedicated tab to display the log messages for the selected Kubernetes resource.
 
 ![logs](images/logs.png)
 
+#### Open Terminals
```
Comment (mohitsuman) [inline README.md:123]:
```suggestion
#### Open Terminal Shell To Running Container
```

File: README.md [hunk 4]
```diff
@@ -112,13 +113,21 @@ The very same dialog may also be opened with a link that's present in the notifi
...
+
+You can open a terminal to a container that's running on the connected cluster.
```
Comment (mohitsuman) [inline README.md:125]:
```suggestion
The plugin allows you to open a terminal shell for the running container on the connected cluster.
You can open a terminal to a container that's running on the connected cluster.
```

File: README.md [hunk 5]
```diff
@@ -112,13 +113,21 @@ The very same dialog may also be opened with a link that's present in the notifi
...
+You select a resource such as a Pod, a Job or a Build and choose "Terminal" in the context menu.
```
Comment (mohitsuman) [inline README.md:126]: The user can open the Kubernetes resources such as Pod, Build and open that in the Terminal Shell using the context menu action `Terminal`.

File: README.md [hunk 6]
```diff
@@ -112,13 +113,21 @@ The very same dialog may also be opened with a link that's present in the notifi
...
+The terminal tab then allows you to switch between the containers of your resource.
```
Comment (mohitsuman) [inline README.md:127]:
```suggestion
All the containers for your Kubernetes resources are listed down in the Terminal View and the user can select the appropriate container and switch between them as needed.
```

File: README.md [hunk 7]
```diff
@@ -112,13 +113,21 @@ The very same dialog may also be opened with a link that's present in the notifi
...
+
+![logs](images/terminal.png)
+
 ## Data and Telemetry
 The JetBrains IntelliJ Kubernetes plugin collects anonymous [usage data](https://github.com/redhat-developer/intellij-kubernetes/blob/master/USAGE_DATA.md) and sends it to Red Hat servers to help improve our products and services.
```
Comment (mohitsuman) [inline README.md:132]:
```suggestion
This `Kubernetes by Red Hat` plugin collects anonymous [usage data](https://github.com/redhat-developer/intellij-kubernetes/blob/main/USAGE_DATA.md) and sends it to Red Hat servers to help improve our products and services.
```

File: src/main/resources/META-INF/plugin.xml [hunk 8]
```diff
@@ -10,6 +10,17 @@
     </description>
     <change-notes>
         <![CDATA[
+        <h3>0.7.0</h3>
+        <ul>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/214">As a user I want to be able to open a terminal in a pod</a></li>
```
Comment (mohitsuman) [inline src/main/resources/META-INF/plugin.xml:15]:
```suggestion
            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/214">Open Terminal Shell for running containers</a></li>
```

File: src/main/resources/META-INF/plugin.xml [hunk 9]
```diff
@@ -10,6 +10,17 @@
...
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/461">cannot follow logs for a pod with multiple containers</a></li>
```
Comment (mohitsuman) [inline src/main/resources/META-INF/plugin.xml:16]:
```suggestion
            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/461">Fix follow logs workflow for a pod with multiple containers</a></li>
```

File: src/main/resources/META-INF/plugin.xml [hunk 10]
```diff
@@ -10,6 +10,17 @@
...
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/492">when switching current namespace, editor still saves to previous namespace</a></li>
```
Comment (mohitsuman) [inline src/main/resources/META-INF/plugin.xml:17]:
```suggestion
            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/492">When switching the current namespace, the editor now saves to the active namespace</a></li>
```

File: src/main/resources/META-INF/plugin.xml [hunk 11]
```diff
@@ -10,6 +10,17 @@
...
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/493">IDE stuck for some second when opening yaml files and not being connected to any cluster</a></li>
```
Comment (mohitsuman) [inline src/main/resources/META-INF/plugin.xml:18]:
```suggestion
            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/493">Fix: IDE stuck for some second when opening YAML files and not being connected to any cluster</a></li>
```

File: src/main/resources/META-INF/plugin.xml [hunk 12]
```diff
@@ -10,6 +10,17 @@
...
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/513">Cannot set new context if there's no current context yet</a></li>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/485">Pushing a 2nd time errors with empty error details</a></li>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/469">lighten up icons and provide separate icons for the light theme</a></li>
```
Comment (mohitsuman) [inline src/main/resources/META-INF/plugin.xml:21]:
```suggestion
            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/469">Improve iconography for light theme</a></li>
```

File: src/main/resources/META-INF/plugin.xml [hunk 13]
```diff
@@ -10,6 +10,17 @@
...
     <change-notes>
         <![CDATA[
+        <h3>0.7.0</h3>
+        <ul>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/214">As a user I want to be able to open a terminal in a pod</a></li>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/461">cannot follow logs for a pod with multiple containers</a></li>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/492">when switching current namespace, editor still saves to previous namespace</a></li>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/493">IDE stuck for some second when opening yaml files and not being connected to any cluster</a></li>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/513">Cannot set new context if there's no current context yet</a></li>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/485">Pushing a 2nd time errors with empty error details</a></li>
```
Comment (mohitsuman) [inline src/main/resources/META-INF/plugin.xml:20]:
```suggestion
            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/485">Fix: Pushing a 2nd time errors with empty error details</a></li>
```

File: src/main/resources/META-INF/plugin.xml [hunk 14]
```diff
@@ -10,6 +10,17 @@
     </description>
     <change-notes>
         <![CDATA[
+        <h3>0.7.0</h3>
+        <ul>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/214">As a user I want to be able to open a terminal in a pod</a></li>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/461">cannot follow logs for a pod with multiple containers</a></li>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/492">when switching current namespace, editor still saves to previous namespace</a></li>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/493">IDE stuck for some second when opening yaml files and not being connected to any cluster</a></li>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/513">Cannot set new context if there's no current context yet</a></li>
```
Comment (mohitsuman) [inline src/main/resources/META-INF/plugin.xml:19]:
```suggestion
            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/513">Fix: Cannot set new context if there's no current context yet</a></li>
```

**Review by mohitsuman — CHANGES_REQUESTED** [review]
Added the suggestions for the wordings.

**Comment by mohitsuman** [comment]
@adietish We should move the PR out of Draft state as it is ready for review.

**Comment by adietish** [comment]
@mohitsuman not yet, I'll turn it ready once we have all fixes merged that we require for the release. There's 1 left: #523

File: README.md (hunk 4 above)
Comment (adietish) [reply README.md:125]: 2nd line is most likely a leftover

File: src/main/resources/META-INF/plugin.xml (hunk 8 above)
Comment (adietish) [reply src/main/resources/META-INF/plugin.xml:15]: @mohitsuman this is not explanatory text. These are the changelog items. There's nothing to correct here.

File: src/main/resources/META-INF/plugin.xml (hunk 9 above)
Comment (adietish) [reply src/main/resources/META-INF/plugin.xml:16]: dito

File: src/main/resources/META-INF/plugin.xml (hunk 11 above)
Comment (adietish) [reply src/main/resources/META-INF/plugin.xml:18]: dito

File: src/main/resources/META-INF/plugin.xml (hunk 13 above)
Comment (adietish) [reply src/main/resources/META-INF/plugin.xml:20]: dito

File: src/main/resources/META-INF/plugin.xml (hunk 14 above)
Comment (adietish) [reply src/main/resources/META-INF/plugin.xml:19]: dito

File: src/main/resources/META-INF/plugin.xml (hunk 12 above)
Comment (adietish) [reply src/main/resources/META-INF/plugin.xml:21]: dito

File: src/main/resources/META-INF/plugin.xml (hunk 8 above)
Comment (adietish) [reply src/main/resources/META-INF/plugin.xml:15]: There's nothing to change here. These are the changelog items which are copy/pasted from the github issues.

File: src/main/resources/META-INF/plugin.xml (hunk 10 above)
Comment (adietish) [reply src/main/resources/META-INF/plugin.xml:17]: dito

File: src/main/resources/META-INF/plugin.xml (hunk 9 above)
Comment (adietish) [reply src/main/resources/META-INF/plugin.xml:16]: dito

**Comment by adietish** [comment]
@jeffmaury, @lstocchi: please review. I will merge this monday, when I do the release.

**Review by lstocchi — CHANGES_REQUESTED** [review]
Can you please bump the common library to 1.8.0? Just released it today .... and I need it otherwise there is an error with the tekton plugin because it ends up invoking the 1.8.0-SNAPSHOT bundled here :/

File: README.md [hunk 15]
```diff
@@ -112,15 +113,25 @@ The very same dialog may also be opened with a link that's present in the notifi
...
 You may delete any resource that is listed in the resource tree by choosing "Delete" in the context menu.
 
-#### Follow logs
+#### Follow Logs of Running Containers
 
-IntelliJ-Kubernetes extension allows you to follow logs of the cluster resources such as Pods, Jobs and Builds.
+IntelliJ Kubernetes extension allows you to follow logs of the cluster resources such as Pods, Jobs and Builds.
 There is a dedicated tab to display the log messages for the selected Kubernetes resource.
+All the containers for your Kubernetes resources are listed down in the Terminal View and the user can select the appropriate container and switch between them as needed.
+Init containers, that are run before the app containers, are listed and can be selected, too.
```
Comment (mohitsuman) [inline README.md:121]:
```suggestion
This also lists down the `init` containers also user can select to see the logs. 
```

File: README.md [hunk 16]
```diff
@@ -23,6 +23,7 @@ It is available for install from the [JetBrains Marketplace](https://plugins.jet
  - Navigate into Kubernetes workloads, services, pods and nodes
  - Switching contexts and namespaces within Kubernetes
  - Follow logs from the connected cluster resource
+ - Open a Terminal Shell to a Running Container in the connected cluster
```
Comment (mohitsuman) [inline README.md:27]:
```suggestion
Support [init containers](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) workflow. These are specialized containers that run before app containers in a [Pod](https://kubernetes.io/docs/concepts/workloads/pods/). Init containers can contain utilities or setup scripts not present in an app image
```

Comment (adietish) [reply README.md:27]: To me these bullet points should be key points of what's possible. You added 3 full sentences. I'd move the explanations further down to he text even though there's already text about it in follow logs and terminal.

File: gradle.properties [hunk 17]
```diff
@@ -2,11 +2,11 @@ ideaVersion=IC-2021.1
 # build number ranges
 # https://www.jetbrains.org/intellij/sdk/docs/basics/getting_started/build_number_ranges.html
 sinceIdeaBuild=211
-projectVersion=0.6.3-SNAPSHOT
+projectVersion=0.7.0
 jetBrainsToken=invalid
 jetBrainsChannel=stable
 intellijPluginVersion=1.9.0
-intellijCommonVersion=1.8.0-SNAPSHOT
+intellijCommonVersion=1.8.0
```
Comment (adietish) [inline gradle.properties:9]: @lstocchi using intellij-common 1.8.0 now.

File: README.md [hunk 18]
```diff
@@ -112,15 +114,25 @@ The very same dialog may also be opened with a link that's present in the notifi
...
 You may delete any resource that is listed in the resource tree by choosing "Delete" in the context menu.
 
-#### Follow logs
+#### Follow Logs of Running Containers
 
-IntelliJ-Kubernetes extension allows you to follow logs of the cluster resources such as Pods, Jobs and Builds.
+IntelliJ Kubernetes extension allows you to follow logs of the cluster resources such as Pods, Jobs and Builds.
 There is a dedicated tab to display the log messages for the selected Kubernetes resource.
+All the containers for your Kubernetes resources are listed down in the Terminal View and the user can select the appropriate container and switch between them as needed.
+This also lists down the `init` containers also user can select to see the logs.
```
Comment (mohitsuman) [inline README.md:122]:
```suggestion
This also lists down the `init` containers which the user can select to see the logs. 
```

File: README.md (hunk 16 above)
Comment (mohitsuman) [reply README.md:28]: I think this work with bullet point.

File: README.md [hunk 19]
```diff
@@ -112,15 +114,25 @@ The very same dialog may also be opened with a link that's present in the notifi
...
+All the containers for your Kubernetes resources are listed down in the Terminal View and the user can select the appropriate container and switch between them as needed.
+This also lists down the `init` containers also user can select to see the logs. 
 
 ![logs](images/logs.png)
 
+#### Open Terminal Shell To Running Container
+The plugin allows you to open a terminal shell for the running container on the connected cluster.
+The user can open the Kubernetes resources such as Pod, Build and open that in the Terminal Shell using the context menu action Terminal.
+All the containers for your Kubernetes resources are listed down in the Terminal View and the user can select the appropriate container and switch between them as needed.
+Init containers, that are run before the app containers, are listed and can be selected, too.
```
Comment (mohitsuman) [inline README.md:130]:
```suggestion
Init containers, which run before the application containers, are listed down and can be selected to open in the terminal shell.
```

File: README.md [hunk 20]
```diff
@@ -112,15 +114,25 @@ The very same dialog may also be opened with a link that's present in the notifi
...
+IntelliJ Kubernetes extension allows you to follow logs of the cluster resources such as Pods, Jobs and Builds.
 There is a dedicated tab to display the log messages for the selected Kubernetes resource.
+All the containers for your Kubernetes resources are listed down in the Terminal View and the user can select the appropriate container and switch between them as needed.
+This also lists down the `init` containers also user can select to see the logs. 
 
 ![logs](images/logs.png)
 
+#### Open Terminal Shell To Running Container
+The plugin allows you to open a terminal shell for the running container on the connected cluster.
+The user can open the Kubernetes resources such as Pod, Build and open that in the Terminal Shell using the context menu action Terminal.
```
Comment (mohitsuman) [inline README.md:128]:
```suggestion
The user can open the Kubernetes resources such as Pod, Build in the Terminal Shell using the context menu action Terminal.
```

**Review by mohitsuman — CHANGES_REQUESTED** [review]
Added some wording changes.

File: README.md (hunk 16 above)
Comment (adietish) [reply README.md:28]: I think that this is far too long for an overview. I especially notice this when reading all the bullet points. It's striking how these are very short but the last one is very verbose.

File: src/main/resources/META-INF/plugin.xml [hunk 21]
```diff
@@ -10,6 +10,19 @@
     </description>
     <change-notes>
         <![CDATA[
+        <h3>0.7.0</h3>
+        <ul>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/214">As a user I want to be able to open a terminal in a pod</a></li>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/461">cannot follow logs for a pod with multiple containers</a></li>
```
Comment (lstocchi) [inline src/main/resources/META-INF/plugin.xml:16]: Maybe to have the first letter as capital as the other points

File: src/main/resources/META-INF/plugin.xml [hunk 22]
```diff
@@ -10,6 +10,19 @@
...
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/523">Support init containers</a></li>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/492">when switching current namespace, editor still saves to previous namespace</a></li>
```
Comment (lstocchi) [inline src/main/resources/META-INF/plugin.xml:18]: same as above

File: src/main/resources/META-INF/plugin.xml [hunk 23]
```diff
@@ -10,6 +10,19 @@
...
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/530">Can push yml resource without a name but no resource is created, "pull" and "diff" actions do nothing</a></li>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/493">IDE stuck for some second when opening yaml files and not being connected to any cluster</a></li>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/513">Cannot set new context if there's no current context yet</a></li>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/485">Pushing a 2nd time errors with empty error details</a></li>
+            <li><a href="https://github.com/redhat-developer/intellij-kubernetes/issues/469">lighten up icons and provide separate icons for the light theme</a></li>
```
Comment (lstocchi) [inline src/main/resources/META-INF/plugin.xml:23]: same as above

**Review by lstocchi — APPROVED** [review]
LGTM, just a tiny styling enhancement for the marketplace page

File: src/main/resources/META-INF/plugin.xml (hunk 21 above)
Comment (adietish) [reply src/main/resources/META-INF/plugin.xml:16]: This is the changelog. I take the exact wording of the issue and copy it over here, 1:1. I see no point in correcting this here.

File: src/main/resources/META-INF/plugin.xml (hunk 8 above)
Comment (mohitsuman) [reply src/main/resources/META-INF/plugin.xml:15]: Yeah but this is what gets displayed on the landing page. So it need not be the GitHub issue same wordings. We need to add wording which explains what functionality it supports now, given it already has an issue link for folks to read more.

**Review by mohitsuman — COMMENTED** [review]
Thinking around the changed log wordings.

File: src/main/resources/META-INF/plugin.xml (hunk 21 above)
Comment (lstocchi) [reply src/main/resources/META-INF/plugin.xml:16]: This text is the one that users see in the marketplace https://plugins.jetbrains.com/plugin/15921-kubernetes-by-red-hat . Up to you what to do but for me it should be taken care a bit. It's like when you write a text and you don't pay attention with punctuation. It's ok if you're in a chat, a bit less for something formal. 
I also saw Mohit comments for the wording. I suck at english so i cannot give any opinion on it but if there is a way to explain better what's the improvement, then it's a change to make for me.  Sticking with the idea that it has to be 1:1 as the changelog is counterproductive from my pov. This is your main ads.
Just my 2 cent. :slightly_smiling_face:

File: src/main/resources/META-INF/plugin.xml (hunk 8 above)
Comment (adietish) [reply src/main/resources/META-INF/plugin.xml:15]: ideally these would be generated automatically, so I'd be cautious with changes. If changes, then those should be done to the github items like we do ex. in tekton where it's required to prefix an issue with "fix:", "feat:", etc. (see here: https://github.com/redhat-developer/intellij-tekton/commits/main).

Comment (mohitsuman) [reply src/main/resources/META-INF/plugin.xml:15]: okay. Let's iterate over this for the next release.

Comment (adietish) [reply src/main/resources/META-INF/plugin.xml:15]: @mohitsuman dare to please +1 then?

**Comment by adietish** [comment]
@jeffmaury: please review and approve/refute. I'd like to release today.

**Review by mohitsuman — APPROVED** [review]
LGTM

**Pull request merged**